tensorstore/internal/http/http_request.h

```diff
@@ -88,19 +101,22 @@ class HttpRequestBuilder {
   /// response.
   HttpRequestBuilder& EnableAcceptEncoding();
 
+  /// Adds a `range` header to the http request if the byte_range
+  /// is specified.
+  HttpRequestBuilder& MaybeAddRangeHeader(OptionalByteRangeRequest byte_range);
+  /// Adds a `cache-control` header specifying `max-age` or `no-cache`.
+  HttpRequestBuilder& MaybeAddCacheControlMaxAgeHeader(absl::Duration max_age);
+  /// Adds a `cache-control` header consistent with `staleness_bound`.
+  HttpRequestBuilder& MaybeAddStalenessBoundCacheControlHeader(absl::Time staleness_bound);
+
```

Comment:
We can just add a `std::vector<std::string> &headers()` accessor.  Then all of these header adding methods can just be changed to operate on a `std::vector<std::string>` and they don't need to be members, and the forwarding in S3RequestBuilder can also be avoided.

S3RequestBuilder can then also just access the headers in its `BuildRequest` method rather than needing to override `AddHeader`.